### PR TITLE
[#183] unification of all works we done

### PR DESCRIPTION
**Pull request by dodoels**
[#183]

- [x] refactor the directory structure: we are currently having files everywhere and not organized. This is to organize the codes in a hierarchical order. 
- [x] make sure the dependencies and logics dont break
- [x] naming convention (distinguish terms, i.e. fact/decision/precedent/binary/text/model)
- [x] unify coding style (OOP/functional, abstract functions, quote format)
- [x] run the pre-processing, clustering, post-processing, and training in a pipeline
- [x] unit test 

New file structure:
```

----| data <all data input and output>
--------| raw <extract the raw data here>
--------| binary <all saved binarized model/data>
--------| cache <temp files>
--------| cluster
------------| fact
------------| decision

----| feature_extraction <all data manipulation before supervised training>
--------| clustering                                           
------------| dbscan                                          
------------| hdbscan
------------| k_means
------------| optimization
------------| clustering.py <driver for clustering>
--------| pre_processing
------------| precedent_model
------------| regex_parse
------------| word_vector
------------| pre_processing.py <driver for pre_processing>
------------| pre_processor.py
--------| post_processing
------------| precedent_vector
------------| post_processing.py <driver for post_processing>
--------| feature_extraction.py <driver for feature extraction (using 3 drivers above)>

----| model_learning <supervised training>
------------| neural_net
------------| model_training.py <driver for supervised training>

----| util <common tool>
------------| log.py <logging tool>
------------| file.py <file save and load>
------------| constant.py <global variables>

init.py
main.py <driver for the pipeline (feature extraction + model training>

```

### CODE REVIEW

**Comment by dodoels** [comment]
please ignore all test files for now

**Comment by Samuel-Campbell** [comment]
Helping bruce with this on wednesday. Not ready for merge

**Review by Samuel-Campbell — APPROVED** [review]
looks good

File: src/ml_service/feature_extraction/post_processing/post_processing.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from feature_extraction.post_processing.precedent_vector.precedent_vector import PrecedentVector
+from util.file import Load, Save
+
+def run ():
+    fact_cluster_model = Load.load_binary("facts_cluster_model.bin")
+    decision_cluster_model = Load.load_binary("decisions_cluster_model.bin")
+    fact_data_tuple = Load.load_binary("facts_pre_processed.bin")
```
Comment (arekmano) [inline src/ml_service/feature_extraction/post_processing/post_processing.py:7]: no more load_binary please. all should be done in one shot

File: src/ml_service/outcome_predictor/basic_neural_net.py [hunk 2]
```diff
@@ -1,52 +0,0 @@
-from keras.models import Sequential
```
Comment (arekmano) [inline src/ml_service/outcome_predictor/basic_neural_net.py:1]: why is this deleted?

**Review by arekmano — CHANGES_REQUESTED** [review]

File: src/ml_service/outcome_predictor/basic_neural_net.py (hunk 2 above)
Comment (dodoels) [reply src/ml_service/outcome_predictor/basic_neural_net.py:1]: We thought we are not using it anymore, do you want me to put it back?

File: src/ml_service/feature_extraction/post_processing/post_processing.py (hunk 1 above)
Comment (dodoels) [reply src/ml_service/feature_extraction/post_processing/post_processing.py:7]: You need to load binaries to create precedent vector lol

File: src/ml_service/outcome_predictor/basic_neural_net.py (hunk 2 above)
Comment (Samuel-Campbell) [reply src/ml_service/outcome_predictor/basic_neural_net.py:1]: it's also not unittested. Add later when you test it

File: src/ml_service/feature_extraction/post_processing/post_processing.py (hunk 1 above)
Comment (Samuel-Campbell) [reply src/ml_service/feature_extraction/post_processing/post_processing.py:7]: I don't understand what you mean by no more binary

**Review by arekmano — APPROVED** [review]

**Pull request merged**